doc.go

```diff
@@ -1,5 +1,5 @@
 /*
-Package godsfapi does not contain any code but is the base for all
+Package dsf does not contain any code but is the base for all
 implementations regarding DuetAPIClient.
 */
-package godsfapi
+package dsf
```

Comment:
Don't name the top-level package other than the GitHub repo name, i.e. `dsf-go` in this case. Same goes for all other locations of top-level `doc.go` files.

Reply:
so either rename it here to dsf_go or i can move everything into a subfolder called dsf?

Reply:
Change the above line to dsf-go. That's effectively the package name anyway, isn't it?

Reply:
moved it into a subfolder. now things are fine.

Reply:
Not exactly what I would have done but that surely works.

Reply:
no dashes allowed in package names.

Reply:
I'm sorry, I messed something up here. You could have left it how it initially was.

Reply:
how come? the compiler complains actually if the directory name doesn't match the package name.

Reply:
There are other prominent libraries (e.g. https://github.com/nats-io/nats.go) that have a package name (`nats` in this case) that differs from the actual repo name. I had in mind that in this case you would have to use a rather wieldy `import` syntax but I must have mixed up something in my head here since that still works.

And I cannot reproduce the compiler complaining. Just tried it myself. Either way, it is probably totally unimportant how the top-level package is named since it only contains `doc.go` as a placeholder so you can find some information on this library at https://pkg.go.dev without having to click on a sub-package first.